File: apiEZPZ/src/main/java/com/example/apiezpz/search/entity/WeeklyRank.java
package com.example.apiezpz.search.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "weekly_rank")
public class WeeklyRank {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String category;

    @Column(nullable = false)
    private Long searchCount;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false)
    private LocalDate recordedDate;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/search/entity/WeeklyRank.kt
package com.example.apiezpz.search.entity

import jakarta.persistence.*
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
@Table(name = "weekly_rank")
data class WeeklyRank(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var category: String = "",

    @Column(nullable = false)
    var searchCount: Long = 0L,

    @Column(nullable = false)
    var startDate: LocalDate = LocalDate.now(),

    @Column(nullable = false)
    var endDate: LocalDate = LocalDate.now(),

    @Column(nullable = false)
    var recordedDate: LocalDate = LocalDate.now(),

    var createdAt: LocalDateTime? = null
) {
    @PrePersist
    protected fun onCreate() {
        this.createdAt = LocalDateTime.now()
    }
} 